master/internal/scheduler/default_resource_provider.go

```diff
@@ -116,13 +113,18 @@ func (d *DefaultRP) assignTask(task *Task) bool {
 		return false
 	}
 
-	d.assigmentByHandler[task.handler] = make([]containerAssignment, 0, len(fits))
+	assignments := make([]Assignment, 0, len(fits))
 
 	for _, fit := range fits {
-		d.assignContainer(task, fit.Agent, fit.Slots, len(fits))
+		assignments = append(
+			assignments,
+			d.assignContainer(task, fit.Agent, fit.Slots, len(fits)),
+		)
 	}
 
-	task.handler.System().Tell(task.handler, TaskAssigned{NumContainers: len(fits)})
+	task.handler.System().Tell(task.handler, TaskAssigned{
```

Comment:
I think they are the same.

Reply:
I didn't get around to reviewing this, but I don't think there's a `ctx` to use here. I think technically this makes it look like the system is sending the message not the RP, so we ever tried to trace the actor system, it'd be good to keep in mind that it might not work that well.

Reply:
@stoksc yeah you are right. the sender would be nil if the system sends the message.